src/core/send.c

```diff
@@ -1120,6 +1121,16 @@ QuicSendFlush(
         return TRUE;
     }
 
+    //
+    // Connection CID changes on idle state after an amount of time
+    //
+    if (Connection->Settings.IdleSrcCidChangeMs != 0 &&
+        Send->LastFlushTimeValid &&
+        CxPlatTimeDiff64(Send->LastFlushTime, TimeNow) >= MS_TO_US(Connection->Settings.IdleSrcCidChangeMs)) {
+        QuicConnGenerateNewSourceCids(Connection, TRUE);
+        Connection->Stats.Misc.SrcCidUpdateCount++;
```

Comment:
This is still incrementing even if we don't use source CIDs (i.e. `!Connection->State.ShareBinding`). I wonder if we should put `Connection->Stats.Misc.SrcCidUpdateCount++` **in** `QuicConnGenerateNewSourceCids`.

Reply:
We can do that, I think. Another thing here is we don't even need to check time if we're not using source CIDs, so I can simply add a condition to if `Connection->State.ShareBinding` like this

Reply:
Oh, on the second thought, I don't think we can do that cause we're using that function for initial generations as well. Reverting and only adding `Connection->State.ShareBinding` to if condition

Reply:
I'll resolve this comment when we agreed on the same thought.

Reply:
Let's move it to `QuicConnGenerateNewSourceCids` but only increment the counter when `ReplaceExistingCids == TRUE`, so it won't be triggered by the initial generation.

Reply:
Done